sdkproject/Assets/Mapbox/Unity/MeshGeneration/Factories/LowPolyTerrainFactory.cs

```diff
@@ -0,0 +1,517 @@
+					_newNormalList.Add(Unity.Constants.Math.Vector3Up);
+
+
+					_newUvList.Add(new Vector2(x / cap, 1 - y / cap));
+					_newUvList.Add(new Vector2((x + 1) / cap, 1 - y / cap));
+					_newUvList.Add(new Vector2(x / cap, 1 - (y + 1) / cap));
+					//--
+					_newUvList.Add(new Vector2((x + 1) / cap, 1 - y / cap));
+					_newUvList.Add(new Vector2((x + 1) / cap, 1 - (y + 1) / cap));
+					_newUvList.Add(new Vector2(x / cap, 1 - (y + 1) / cap));
+
+					_newTriangleList.Add(triStart);
+					_newTriangleList.Add(triStart + 1);
+					_newTriangleList.Add(triStart + 2);
+					//--
+					_newTriangleList.Add(triStart + 3);
+					_newTriangleList.Add(triStart + 4);
+					_newTriangleList.Add(triStart + 5);
+				}
+			}
+
+
+			var mesh = tile.MeshFilter.mesh;
+			mesh.SetVertices(_newVertexList);
+			mesh.SetNormals(_newNormalList);
+			mesh.SetUVs(0, _newUvList);
+			mesh.SetTriangles(_newTriangleList, 0);
+			mesh.RecalculateBounds();
+		}
+
+		internal override void OnUnregistered(UnityTile tile)
+		{
+			_meshData.Remove(tile.CanonicalTileId);
+		}
+
+		/// <summary>
+		/// Creates the non-flat terrain using a height multiplier
+		/// </summary>
+		/// <param name="tile"></param>
+		/// <param name="heightMultiplier">Multiplier for queried height value</param>
+		private void CreateTerrainHeight(UnityTile tile)
+		{
+			tile.HeightDataState = TilePropertyState.Loading;
+			var pngRasterTile = new RawPngRasterTile();
+
+			tile.AddTile(pngRasterTile);
+			Progress++;
+
+			pngRasterTile.Initialize(_fileSource, tile.CanonicalTileId, _mapId, () =>
+			{
+				if (pngRasterTile.HasError)
+				{
+					tile.HeightDataState = TilePropertyState.Error;
+
+					// Handle missing elevation from server (404)!
+					// TODO: optimize this search!
+					if (pngRasterTile.ExceptionsAsString.Contains("404"))
+					{
+						ResetToFlatMesh(tile);
+					}
+					Progress--;
+					return;
+				}
+
+				tile.SetHeightData(pngRasterTile.Data, _heightModifier);
+				GenerateTerrainMesh(tile);
+				Progress--;
+			});
+		}
+
+		/// <summary>
+		/// Creates the non-flat terrain mesh, using a grid by defined resolution (_sampleCount). Vertex order goes right & up. Normals are calculated manually and UV map is fitted/stretched 1-1.
+		/// Any additional scripts or logic, like MeshCollider or setting layer, can be done here.
+		/// </summary>
+		/// <param name="tile"></param>
+		/// <param name="heightMultiplier">Multiplier for queried height value</param>
+		private void GenerateTerrainMesh(UnityTile tile)
+		{
+#if UNITY_5_5_OR_NEWER
+            tile.MeshFilter.mesh.GetVertices(_currentTileMeshData.Vertices);
+            tile.MeshFilter.mesh.GetNormals(_currentTileMeshData.Normals);
+#else
+			_currentTileMeshData.Vertices.Clear();
+			_currentTileMeshData.Vertices.AddRange(tile.MeshFilter.mesh.vertices);
+			_currentTileMeshData.Normals.Clear();
+			_currentTileMeshData.Normals.AddRange(tile.MeshFilter.mesh.normals);
```

Comment:
You need to recalculate bounds so meshes don't disappear from the camera's view.